core/src/main/scala/akka/kafka/internal/BaseSingleSourceLogic.scala

```diff
@@ -17,7 +18,12 @@ import org.apache.kafka.common.TopicPartition
 import scala.annotation.tailrec
 import scala.concurrent.{ExecutionContext, Future}
 
-private[kafka] abstract class BaseSingleSourceLogic[K, V, Msg](
+/**
+ * Internal API.
+ *
+ * Shared GraphStageLogic for [[SingleSourceLogic]] and [[ExternalSingleSourceLogic]].
+ */
+@InternalApi private abstract class BaseSingleSourceLogic[K, V, Msg](
```

Comment:
The `Internal API` api doc is mostly relevant on code which show up in the generated scaladoc or javadoc. As `private[package]` members still show in in javadoc, this is where `Internal API` is mostly useful.

Here, and below, by making these `private[kafka]` classes `private` we can remove the comment as well.